### PR TITLE
Atmos Grenades

### PR DESCRIPTION
**Pull request by ghost**
From https://github.com/ParadiseSS13/Paradise/pull/5132

Adds Atmos Grenades to the uplink, 3 clusterbuster grenades (plasma, freon and n2o) in a box for 11 TC.

:cl:
add: Atmos grenades are now in the game and uplink
/:cl:

### CODE REVIEW

**Comment by A-t48** [comment]
Not freon? 0/10

**Comment by uraniummeltdown** [comment]
>Not freon? 0/10

Freon sounds neat will have to bump up the price to 8 TC then, but I'm tired right now so I'll add it and test tomorrow

File: code/game/objects/items/weapons/grenades/atmosgrenade.dm [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+/obj/item/weapon/grenade/gas/knockout
+	name = "Knockout Grenade"
+	desc = "A grenade that completely removes all air and heat from its detonation area."
+
+/obj/item/weapon/grenade/gas/knockout/prime()
+	var/turf/target_turf = get_turf(src)
+	if(istype(target_turf))
+		target_turf.atmos_spawn_air("n2o=100;TEMP=[T20C]")
+		target_turf.air_update_turf()
+	qdel(src)
```
Comment (phil235) [inline code/game/objects/items/weapons/grenades/atmosgrenade.dm]: try to find a way to have both grenade use the same prime(). This is pretty much a copypasta.

**Comment by Signmark** [comment]
https://github.com/ParadiseSS13/Paradise/pull/5646
Also look at another PR

**Comment by PKPenguin321** [comment]
yeah i was gonna say, this might be a bit too good, especially with how much damage regular clothing takes from fires now

File: code/game/objects/items/weapons/grenades/atmosgrenade.dm (hunk 1 above)
Comment (Shadowlight213) [inline code/game/objects/items/weapons/grenades/atmosgrenade.dm]: Could do a var on each grenade that's the argument put into atmos_spawn air()

**Comment by AdamElTablawy** [comment]
>pocket plasmaflood grenade on demand

>pocket aoe permafuck grenade on demand

>cheap as dick

Really

Fucking no

Hell more to the point why would the syndicate have plasma seeing as one of the many reasons they hate NT is because they're hogging all the plasma

**Comment by swindly** [comment]
The plasma grenades aren't too powerful. 100 units of phlogiston has the exact same effect.

**Comment by KingPhilipIII** [comment]
Keep on mind Paradise doesn't have the burn all your shit pull request. It would be considerably more powerful here.

**Comment by uraniummeltdown** [comment]
Here's the plasma grenade used in the Bar on Paradise and then on TG. Paradise's Bar has reinforced windows. You can see why they increased the cost on Paradise, it does much more area damage.
paraplasma
tgplasma

It's hardly a plasmaflood grenade.

**Comment by KingPhilipIII** [comment]
Oh right. Forgot fire can melt walls on Para. They still don't have the issue of nonflammable items melting however, so if they lob a grenade in medbay most of the stuff is still standing. Here all the medical supplies burn.

**Comment by uraniummeltdown** [comment]
I'll increase the price to 12 TC, going to add the Freon grenade

File: code/game/objects/items/weapons/grenades/clusterbuster.dm [hunk 2]
```diff
@@ -126,3 +126,18 @@
...
+ 	payload = /obj/item/weapon/grenade/gas
+
+/obj/item/weapon/grenade/clusterbuster/n2o
+ 	name = "N2O Cluster Grenade"
+ 	desc = "For when you need to knock out EVERYONE."
+ 	payload = /obj/item/weapon/grenade/gas/knockout
+
+/obj/item/weapon/grenade/clusterbuster/freon
+ 	name = "Freon Cluster Grenade"
+ 	desc = "For when you need everyone to chill out."
```
Comment (Iamgoofball) [inline code/game/objects/items/weapons/grenades/clusterbuster.dm]: what killed the dinosaurs

Comment (Shadowlight213) [inline code/game/objects/items/weapons/grenades/clusterbuster.dm]: Freon!
The truth is revealed

**Comment by KingPhilipIII** [comment]
Man. It's gonna be great as an elite nukie with the ability to make a plasmafire on command.

**Comment by ExcessiveUseOfCobblestone** [comment]
> Atmos simulator
> Can't even flood without having a grenade do it for you

DISGUSTING

File: code/game/objects/items/weapons/grenades/clusterbuster.dm (hunk 2 above)
Comment (Niknakflak) [inline code/game/objects/items/weapons/grenades/clusterbuster.dm]: The dinosaurs were killed by a big ball of fire and death that caused cold, not a big ball of cold that caused death.

**Comment by AdamElTablawy** [comment]
yes, you can make a far more effective, cheap, and free plasmaflood grenade simply by going to chemistry, grinding a bar or two of plasma and mixing up a smoke grenade

hell you can be an absolute fuckbitch and use it in an advanced grenade assembly to constantly and endlessly spew massive ass streams of plasma

but more to the point this is still a plasmaflood grenade ON COMMAND, no chemistry access required, wehre before at the least you either had to be a chemist/cmo/hop or break into chemistry (and get sprayed with acid)

so uh you know

this is a really really shit idea on a scale of shit to goofball ideas simply because every round is going to start, people are gonna go to their jobs 'PLASMA IN BRIG' 'PLASMA IN ENGI' 'PLASMA' *THE EMERGENCY SHUTTLE HAS BEEN CALLED*

like seriously do you remember how hilariously deadly plasmafires are, especially after fire buffs, because 'a phlog grenade might be worse' is still not an argument for something that'll fuck the brig up the asshole hard within minutes of roundstart

also no, a phlog grenade is not as bad as this. i've tested. i can go into the exact math of it (yes, math, numbers) if you want me to lay down bit by fucking bit how this would be worse than a phlog grenade by far, not accounting all the other variables that i've mentioned above

and that's just the plasma grenade.

a freon grenade-seriously? you remember how much of a god-awful clench your butthole meme freon is when used to griff, right, because this will very quickly devolve into suffering

and an n2o grenade-n2o takes fucking forever to get rid of unless you jury rig an air pump-volume pump-freezer-to-filter combo, which you'd need an RPD and fucking freezer parts to do. scrubbing it doesn't remove it, you need to use an air pump to drain it out, a volume pump to move it fast, and dump it right into the filter pipes while also making sure the room doesn't run out of air due to the air pump. 

and anything that can be said about plasma, can be worse for n2o, especially because while n2o doesn't ignite, sure, it basically permaKOS anyone in it, which is fine and dandy if someone is stupid enough to walk into the obvious white clouds, but far worse in pocket grenade form where someone sets it off and oh god oh god there goes my round i'm asleep forever and he's stabbing me to death because i was in medbay healing people and got gassed




tl;dr you're shit

**Comment by A-t48** [comment]
Uhh. 

1. Access atmos panel in room
2. Set panel to draught
3. Fuck around with filters to give expanded range
4. Done!

**Comment by A-t48** [comment]
Syndicate bombs are still more powerful than any of these grenades, and aren't as easily escapable. And somewhat permanently fuck an area.

**Comment by KingPhilipIII** [comment]
Atleast atmos techs will have more to do cleaning up the freon and N2O. And it's a fairly expensive item to get that has somewhat limited use outside of assassination. People might use them a lot at first as they play with a new toy but I doubt it will be a major problem.

**Comment by Tacolizard** [comment]
@PKPenguin321 this is why we need #20958

**Comment by AdamElTablawy** [comment]
so you mean like the macrobombs, philip?

remember traitor macrobombs?

honestly these will just be used for endless shuttle grief

**Comment by Iamgoofball** [comment]
How is it griefing to use traitor tools as a traitor

**Comment by AdamElTablawy** [comment]
its not i'm just using shuttle grief as a general descriptor for shit that kills everyone on the shuttle, like bombs or boh bombs or clf3 smoke grenades

File: code/game/objects/items/weapons/grenades/atmosgrenade.dm [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+/obj/item/weapon/grenade/gas/prime()
+	var/turf/target_turf = get_turf(src)
+	if(istype(target_turf))
+		target_turf.atmos_spawn_air("plasma=100;TEMP=1000")
+		target_turf.air_update_turf()
+	qdel(src)
+
+/obj/item/weapon/grenade/gas/knockout
+	name = "Knockout Grenade"
+	desc = "A grenade that completely removes all air and heat from its detonation area."
```
Comment (optimumtact) [inline code/game/objects/items/weapons/grenades/atmosgrenade.dm]: Address this please
